src/librustc_typeck/astconv.rs

```diff
@@ -1424,15 +1424,15 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
                             if err_for_lt { continue }
                             err_for_lt = true;
                             (struct_span_err!(self.tcx().sess, lt.span, E0110,
-                                              "lifetime parameters are not allowed on this type"),
+                                              "lifetime parameters are not allowed on this name"),
```

Comment:
"name" -> "entity" perhaps?
Or, going into another direction, "name" -> "path segment".
Both "type" (plain incorrect) and "name" (generic arguments are not applied to names) don't look good here.

Reply:
Also, since you are introducing the churn already, could you replace "parameters" with "arguments" here and below?

Reply:
@petrochenkov 

1. Fair enough about the tests. Only two failures right now, one of which is actually desired. There is one thing passing that shouldn't. I think you understand, but I agree it could be clearer to help you as the reviewer.

2. Yes, will move cosmetic stuff to a separate PR.

3. "Path segment" is too technical a term to be shown shown to the user, I feel. "Entity" could work though. Sure, I'll replace "parameters" with "arguments" too.

Reply:
Alternatively, we could do `type_of` on the HIR and then use `kind_name` in the error message. Thoughts?

Reply:
>Alternatively, we could do type_of

You can certainly try!
This won't work on modules though, and perhaps associated types as well?
We can always fallback to "entity" though, so any simple heuristic would work.

Reply:
(That said, this PR is about variants on type aliases, so it's not necessary to pull every possible improvement to rustc diagnostics into it.)

Reply:
True!